website/views/options/social/group.jade

```diff
@@ -22,7 +22,7 @@ a.pull-right.gem-wallet(ng-if='group.type!="party"', popover-trigger='mouseenter
                 span.glyphicon.glyphicon-ban-circle
                 =env.t('leave')
               a.btn.btn-success.pull-right(ng-if='!isMemberOfGroup(User.user._id, group)', ng-click='join(group)')=env.t('join')
-            span(ng-if='group.leader._id == user.id')
+            span(ng-if='group.leader._id == user.id || (user.contributor.admin === true && group.type === "guild")')
```

Comment:
Since this logic is now more complex and used multiple places, it should be put onto `$scope` instead in the client controller for the group.